nuxeo-core/nuxeo-core-mongodb/src/test/java/org/nuxeo/ecm/core/mongodb/configuration/TestMongoDBBackEnd.java

```diff
@@ -0,0 +1,161 @@
+import static org.junit.Assert.fail;
+import static org.nuxeo.ecm.core.storage.dbs.DBSDocument.KEY_NAME;
+import static org.nuxeo.ecm.core.storage.dbs.DBSDocument.KEY_PARENT_ID;
+import static org.nuxeo.ecm.core.storage.mongodb.MongoDBRepository.REPOSITORY_CONNECTION_PREFIX;
+import static org.nuxeo.runtime.mongodb.MongoDBFeature.DEFAULT_MONGODB_DBNAME;
+
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+import org.bson.Document;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CloseableCoreSession;
+import org.nuxeo.ecm.core.api.ConcurrentUpdateException;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.security.ACE;
+import org.nuxeo.ecm.core.api.security.ACL;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+import org.nuxeo.ecm.core.api.security.impl.ACPImpl;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.mongodb.MongoDBConnectionService;
+import org.nuxeo.runtime.mongodb.MongoDBFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import com.google.inject.Inject;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+
+/**
+ * Work In Progress: These tests should be separated to Configuration Test (shouldCreateUniqueIndex) and a new once to
+ * check the MongoDBBackend like TestSQLBackend. Or We can create a new class test to make the check on SQL (postgresql)
+ * and DBS (mongo) to discuss with the team. org.nuxeo.ecm.core.storage.mongodb can hold the back end test, but we have
+ * a cyclic maven dependency module nuxeo-core-test depends on org.nuxeo.ecm.core.storage.mongodb to build
+ * {@link org.nuxeo.ecm.core.test.StorageConfiguration} and this once need {@link CoreFeature}.
+ * <p>
+ * {@link CoreFeature} configure / load {@link org.nuxeo.ecm.core.test.StorageConfiguration} we should find a way to
+ * deploy others custom contribution {@link org.nuxeo.ecm.core.storage.mongodb.MongoDBRepositoryService} to easily
+ * enable / disable the contraints name checking in Tests .... perhaps we need to implement merge on the contribution
+ * 
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ MongoDBFeature.class, CoreFeature.class })
+@Deploy("org.nuxeo.ecm.core.mongodb.test:OSGI-INF/mongodb-disable-renaming-listener-test.xml")
+public class TestMongoDBBackEnd {
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    @Inject
+    protected CoreSession coreSession;
+
+    @Test
+    public void shouldCreateUniqueIndex() {
+        MongoDBConnectionService mongoService = Framework.getService(MongoDBConnectionService.class);
+        MongoDatabase database = mongoService.getDatabase(REPOSITORY_CONNECTION_PREFIX + "test");
```

Comment:
I don't think this kind of white-box test is useful. You're basically testing that you correctly wrote code to create the unique index, but this is not that useful. What's useful is to test the _consequence_ of the index, that we have an exception when we attempt to create two identical parentid+childname.
So I'd remove this test.

Reply:
yes i agree with your point view of we `What's useful is to test the consequence of the index,` and i asked myself the same question, but i did it to ensure that the whole process  is correctly handled when i add the contribution.